hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/account/MirrorEvmContractAliases.java

```diff
@@ -17,23 +17,61 @@
 package com.hedera.mirror.web3.evm.account;
 
 import static com.hedera.mirror.common.util.DomainUtils.toEvmAddress;
+import static com.hedera.services.utils.MiscUtils.isRecoveredEvmAddress;
 
 import com.hedera.mirror.common.domain.entity.EntityType;
 import com.hedera.mirror.web3.evm.store.contract.MirrorEntityAccess;
 import com.hedera.mirror.web3.exception.EntityNotFoundException;
 import com.hedera.mirror.web3.exception.InvalidParametersException;
 import com.hedera.node.app.service.evm.accounts.HederaEvmContractAliases;
+import com.hedera.node.app.service.evm.utils.EthSigsUtils;
+import com.hedera.services.jproto.JECDSASecp256k1Key;
+import com.hedera.services.jproto.JKey;
+import edu.umd.cs.findbugs.annotations.Nullable;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.UnaryOperator;
 import lombok.RequiredArgsConstructor;
 import org.apache.tuweni.bytes.Bytes;
 import org.hyperledger.besu.datatypes.Address;
 
 @RequiredArgsConstructor
 public class MirrorEvmContractAliases extends HederaEvmContractAliases {
+
+    public static final UnaryOperator<byte[]> ADDRESS_RECOVERY_FN = EthSigsUtils::recoverAddressFromPubKey;
     final Map<Address, Address> aliases = new HashMap<>();
     private final MirrorEntityAccess mirrorEntityAccess;
 
+    public boolean maybeLinkEvmAddress(@Nullable final JKey key, final Address address) {
+        return maybeLinkEvmAddress(key, address, ADDRESS_RECOVERY_FN);
+    }
+
+    boolean maybeLinkEvmAddress(
+            @Nullable final JKey key, final Address address, final UnaryOperator<byte[]> addressRecovery) {
+        final var evmAddress = tryAddressRecovery(key, addressRecovery);
+        if (isRecoveredEvmAddress(evmAddress)) {
+            link(Address.wrap(Bytes.wrap(evmAddress)), address); // NOSONAR we have a null check
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    @Nullable
+    public static byte[] tryAddressRecovery(@Nullable final JKey key, final UnaryOperator<byte[]> addressRecovery) {
+        if (key != null) {
+            // Only compressed keys are stored at the moment
+            final var keyBytes = key.getECDSASecp256k1Key();
+            if (keyBytes.length == JECDSASecp256k1Key.ECDSA_SECP256K1_COMPRESSED_KEY_LENGTH) {
+                final var evmAddress = addressRecovery.apply(keyBytes);
```

Comment:
Why not just do the below? It's more straightforward and easier to understand.
```suggestion
                final var evmAddress = EthSigsUtils.recoverAddressFromPubKey(keyBytes);
```

Reply:
Fixed